stonesoup/dataassociator/tracktotrack.py

```diff
@@ -91,31 +122,47 @@ def associate_tracks(self, tracks_set_1: Set[Track], tracks_set_2: Set[Track]):
                 # At this point we should have two lists of states from
                 # track1 and 2 only at the times that they both existed
 
-                n_succesful = 0
-                n_unsuccesful = 0
+                n_successful = 0
+                n_unsuccessful = 0
                 start_timestamp = None
                 end_timestamp = None
                 # Loop through every detection pair and form associations
                 for state1, state2 in zip(track1_states, track2_states):
+                    tot = len(state1.state_vector)
+                    if not self.pos_map:
+                        self.pos_map = [i for i in range(0, tot)]
+                    map_len = len(self.pos_map)
+                    if not self.use_positional_only and tot-map_len > 0:
+                        v_weight = (1 - self.position_weighting) / (tot-map_len)
+                        p_weight = self.position_weighting / len(self.pos_map)
+                    else:
+                        p_weight = 1 / len(self.pos_map)
+                        v_weight = 0
+
+                    weights = [p_weight if i in self.pos_map else v_weight
+                               for i in range(tot)]
+
+                    if isinstance(self.measure, EuclideanWeighted):
+                        self.measure = EuclideanWeighted(weighting=weights)
```

Comment:
Slightly hesitant to instantiate a new measure every loop, as this will be slow given large number of iterations. Could the logic be done upfront? The state spaces should be consistent.